workflow-controller/controller-service/src/main/java/io/hyscale/controller/validator/impl/VolumeValidator.java

```diff
@@ -0,0 +1,308 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.controller.validator.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+
+import io.hyscale.commons.constants.K8SRuntimeConstants;
+import io.hyscale.commons.constants.ToolConstants;
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.commons.logger.WorkflowLogger;
+import io.hyscale.commons.models.AnnotationKey;
+import io.hyscale.commons.models.K8sAuthorisation;
+import io.hyscale.commons.models.StorageClassAnnotation;
+import io.hyscale.commons.utils.HyscaleStringUtil;
+import io.hyscale.commons.utils.ResourceSelectorUtil;
+import io.hyscale.commons.validator.Validator;
+import io.hyscale.controller.builder.K8sAuthConfigBuilder;
+import io.hyscale.controller.model.WorkflowContext;
+import io.hyscale.deployer.core.model.ResourceKind;
+import io.hyscale.deployer.services.exception.DeployerErrorCodes;
+import io.hyscale.deployer.services.handler.ResourceHandlers;
+import io.hyscale.deployer.services.handler.ResourceLifeCycleHandler;
+import io.hyscale.deployer.services.handler.impl.V1PersistentVolumeClaimHandler;
+import io.hyscale.deployer.services.handler.impl.V1StorageClassHandler;
+import io.hyscale.deployer.services.model.DeployerActivity;
+import io.hyscale.deployer.services.provider.K8sClientProvider;
+import io.hyscale.deployer.services.util.KubernetesVolumeUtil;
+import io.hyscale.servicespec.commons.fields.HyscaleSpecFields;
+import io.hyscale.servicespec.commons.model.service.ServiceSpec;
+import io.hyscale.servicespec.commons.model.service.Volume;
+import io.kubernetes.client.custom.Quantity;
+import io.kubernetes.client.openapi.ApiClient;
+import io.kubernetes.client.openapi.models.V1PersistentVolumeClaim;
+import io.kubernetes.client.openapi.models.V1ResourceRequirements;
+import io.kubernetes.client.openapi.models.V1StorageClass;
+
+@Component
+public class VolumeValidator implements Validator<WorkflowContext>{
+	private static final Logger logger = LoggerFactory.getLogger(VolumeValidator.class);
+	
+	private static final String STORAGE = "storage";
+
+	@Autowired
+	private K8sClientProvider clientProvider;
+
+	@Autowired
+	private K8sAuthConfigBuilder authConfigBuilder;
+
+	private List<V1StorageClass> storageClassList = new ArrayList<>();
+
+	@Override
+	public boolean validate(WorkflowContext context) throws HyscaleException {
```

Comment:
With the new changes, the user won't know what failed as we are not printing any message. Need to think on how to properly display message to the user.